controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java

```diff
@@ -127,6 +130,19 @@ public boolean authorize(String resource, Principal principal, String credential
         return handler.authorize(resource, principal).ordinal() >= level.ordinal();
     }
 
+
+    /**
+     * This method is not only visible for testing, but also intended to be used solely for testing. It allows tests
+     * to inject and register custom auth handlers. Also, this method is idempotent.
+     *
+     * @param authHandler the {@code AuthHandler} implementation to register
+     * @Throws NullPointerException {@code authHandler} is null
+     */
+    @VisibleForTesting
+    public synchronized void registerHandler(AuthHandler authHandler) {
```

Comment:
If you want to avoid having this method public, you could make it protected and create an extended class for testing.